src/main/java/net/snowflake/ingest/streaming/internal/SnowflakeStreamingIngestClientInternal.java

```diff
@@ -260,6 +261,27 @@ public String getName() {
     return this.name;
   }
 
+  /**
+   * Return the latest committed/persisted offset token for all channels
+   *
+   * @return map of channel to the latest persisted offset token
+   */
+  @Override
+  public Map<SnowflakeStreamingIngestChannel, String> getLatestCommittedOffsetTokens(
+      List<SnowflakeStreamingIngestChannel> channels) {
+    List<SnowflakeStreamingIngestChannelInternal<?>> internalChannels =
+        channels.stream()
+            .map(c -> (SnowflakeStreamingIngestChannelInternal<?>) c)
+            .collect(Collectors.toList());
+    List<ChannelsStatusResponse.ChannelStatusResponseDTO> channelsStatus =
```

Comment:
Should we check the statusCode first?

Reply:
Status Code is checked in `getChannelStatus` https://github.com/jdcaperon/snowflake-ingest-java/blob/069269994c2a8b59aa8b94f954bd0848e3c8ba99/src/main/java/net/snowflake/ingest/streaming/internal/SnowflakeStreamingIngestClientInternal.java#L417-L439

We can check it here as well, what would you like to do if the status is not valid?

Reply:
If it's checked then it's fine